camel/messages/base.py

```diff
@@ -167,37 +164,32 @@ def extract_text_and_code_prompts(
 
         return text_prompts, code_prompts
 
-    def to_openai_message(self, role: Optional[str] = None) -> OpenAIMessage:
+    def to_openai_message(self, role: str) -> OpenAIMessage:
         r"""Converts the message to an :obj:`OpenAIMessage` object.
 
         Args:
-            role (Optional[str]): The role of the message in OpenAI chat
+            role (str): The role of the message in OpenAI chat
                 system, either :obj:`"system"`, :obj:`"user"`, or
-                obj:`"assistant"`. (default: :obj:`None`)
+                obj:`"assistant"`.
 
         Returns:
             OpenAIMessage: The converted :obj:`OpenAIMessage` object.
         """
-        role = role or self.role
         if role not in {"system", "user", "assistant"}:
             raise ValueError(f"Unrecognized role: {role}")
         return {"role": role, "content": self.content}
 
-    def to_openai_chat_message(
-        self,
-        role: Optional[str] = None,
-    ) -> OpenAIChatMessage:
+    def to_openai_chat_message(self, role: str) -> OpenAIChatMessage:
```

Comment:
Do we still need this?

Reply:
I don't know. I am just refactoring. If you think we do not need this, I will delete it.

Reply:
Sure, let's remove it

Reply:
Removing